x/liquidity/keeper/invariants.go

```diff
@@ -88,15 +88,22 @@ func MintingPoolCoinsInvariant(poolCoinTotalSupply, mintPoolCoin, depositCoinA,
 		depositCoinB = depositCoinB.Sub(refundedCoinB)
 	}
 
-	poolCoinRatio := mintPoolCoin.Quo(poolCoinTotalSupply)
-	depositCoinARatio := depositCoinA.Quo(lastReserveCoinA)
-	depositCoinBRatio := depositCoinB.Quo(lastReserveCoinB)
+	poolCoinRatio := mintPoolCoin.ToDec().QuoInt(poolCoinTotalSupply)
+	depositCoinARatio := depositCoinA.ToDec().QuoInt(lastReserveCoinA)
+	depositCoinBRatio := depositCoinB.ToDec().QuoInt(lastReserveCoinB)
+	expectedMintPoolCoinAmtBasedA := depositCoinARatio.MulInt(poolCoinTotalSupply).TruncateInt()
+	expectedMintPoolCoinAmtBasedB := depositCoinBRatio.MulInt(poolCoinTotalSupply).TruncateInt()
 
 	// NewPoolCoinAmount / LastPoolCoinSupply <= AfterRefundedDepositCoinA / LastReserveCoinA
 	// NewPoolCoinAmount / LastPoolCoinSupply <= AfterRefundedDepositCoinA / LastReserveCoinB
 	if depositCoinARatio.LT(poolCoinRatio) || depositCoinBRatio.LT(poolCoinRatio) {
 		panic("invariant check fails due to incorrect ratio of pool coins")
 	}
+
+	if sdk.MaxInt(mintPoolCoin, expectedMintPoolCoinAmtBasedA).Sub(sdk.MinInt(mintPoolCoin, expectedMintPoolCoinAmtBasedA)).Quo(mintPoolCoin).ToDec().GT(errorRateThreshold) ||
+		sdk.MaxInt(mintPoolCoin, expectedMintPoolCoinAmtBasedB).Sub(sdk.MinInt(mintPoolCoin, expectedMintPoolCoinAmtBasedA)).Quo(mintPoolCoin).ToDec().GT(errorRateThreshold) {
+		panic("invariant check fails due to incorrect amount of pool coins")
+	}
```

Comment:
Dividing between `sdk.Int`s then converting the result to `sdk.Dec` and comparing it against `errorRateThreshold` will be a bit ineffective, because it means that the result of the division should be `0` to pass the condition.

Reply:
Also, when we use `errorRateThreshold`, I first checked that the coin amounts are greater than `coinAmountThreshold`. I think we need that check here, too.

Reply:
Thanks for the comments, It is applied the last commit